backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java

```diff
@@ -37,6 +37,80 @@
 
 public class ElasticSearchConfiguration {
 
+    public enum HostScheme {
+        HTTP("http"),
+        HTTPS("https");
+
+        public static Optional<HostScheme> maybe(String schemeAsString) {
```

Comment:
`maybe` is already encoded in the return type. What you lack is the action you perform, here `parseNullable` would fit